src/movies.js

```diff
@@ -1,28 +1,156 @@
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  moviesArray = moviesArray.map((movie) => movie.director);
+
+  const cleanMoviesArr = [];
+  
+  for (let i = 0; i < moviesArray.length; i++) {
+    if (cleanMoviesArr.indexOf(moviesArray[i]) === -1) {
+      cleanMoviesArr.push(moviesArray[i]);
+    }
+  }
+  return cleanMoviesArr
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0;
+  }
+  
+  const movStevenArr = moviesArray.filter(
+    (movie) => movie.director === 'Steven Spielberg'
+  );  
+
+  const result = movStevenArr.filter(
+    (movie) => movie.genre.indexOf(`Drama`) >= 0
+  );
+
+  return result.length
+  
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0
+  }
+
+  const revisedArray = moviesArray.filter((movie) => typeof movie.score === 'number');
+
+  console.log(revisedArray);
+
+  const sum = revisedArray.reduce(
+    (accumulator, currentValue) => accumulator + currentValue.score,
+    0
+  );
+
+  const average = sum / moviesArray.length;
+
+  return Number(average.toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  
+  const dramaArray = moviesArray.filter(
+    (movie) => movie.genre.indexOf('Drama') >= 0
+    );
+    
+    if (dramaArray.length ===0) {
+      return 0;
+    }
+ 
+  const sum = dramaArray.reduce ((accumulator, currentValue) => accumulator + currentValue.score,0);
+  
+  const average = sum / dramaArray.length
+  
+  return Number(average.toFixed(2));
+  
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+  const moviesArrCopy = [...moviesArray];
+
+  const novArrAsc = moviesArrCopy.sort(function (a, b) {
+    return a.year - b.year || a.title.localeCompare(b.title);  //--> note to self: localeCompare a good way to concatenate filters when sorting.
+  });
+  
+  return novArrAsc;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+  const titleArray = [];
+  let number;
+
+  moviesArray.forEach((movie) => titleArray.push(movie.title));
+  
+  const sortedFullArr = titleArray.sort();
+  
+  if (sortedFullArr.length < 20) {
+    number = sortedFullArr.length;
+  } else {
+    number = 20;
+  }
+  
+  sortedArr = sortedFullArr.splice(0, number);
```

Comment:
This is quite bad practice - make sure to always use `const` or `let` when declaring a variable

Reply:
I feel like I need to pracitce scoping :)